bin/run.js

```diff
@@ -50,6 +62,74 @@ module.exports = function run( files, options ) {
 	QUnit.on( "runEnd", function setExitCode( data ) {
 		if ( data.testCounts.failed ) {
 			process.exitCode = 1;
+		} else {
+			process.exitCode = 0;
+		}
+	} );
+};
+
+run.restart = function( args ) {
+	if ( QUnit.config.queue.length ) {
+		console.log( "Finishing current test and restarting..." );
+	} else {
+		console.log( "Restarting..." );
+	}
+
+	run.abort( () => run.apply( null, args ) );
+};
+
+run.abort = function( callback ) {
+	if ( QUnit.config.queue.length ) {
+		const nextTestIndex = QUnit.config.queue.findIndex( fn => fn.name === "runTest" );
+		QUnit.config.queue.splice( nextTestIndex );
+		QUnit.on( "runEnd", function() {
+			delete global.QUnit;
+			QUnit = null;
+			if ( callback ) {
+				callback();
+			}
+		} );
+	} else {
+		delete global.QUnit;
```

Comment:
Any point in extracting this logic to a closure, so that this code isn't duplicated with the above case?